Dockerfile

```diff
@@ -1,46 +1,57 @@
-FROM debian:bookworm-slim AS builder
+FROM alpine:3.20 AS builder
 
-RUN apt-get update && \
-    apt-get install -y \
+RUN apk add --update --no-cache \
         binutils-dev \
-        build-essential \
+        # Debian: build-essential
+        build-base \
         cmake \
         git \
-        libcurl4-openssl-dev \
-        libdw-dev \
-        libiberty-dev \
-        libssl-dev \
+        # Debian: libcurl4-openssl-dev
+        curl-dev \
+        curl-static \
+        # Debian: libdw-dev
+        libdw \
+        # Debian, alpine bundled in binutils-dev
+        # libiberty-dev \
+        # Debian: libssl-dev
+        openssl-dev \
         ninja-build \
         python3 \
-        zlib1g-dev \
-        ;
+        # Debian: zlib1g-dev
+        zlib-dev \
+        # Debian: libelf-dev
+        elfutils-dev \
+        # Debian: libstdc++-12-dev
+        libstdc++-dev \
+    ;
 
 ADD . /src/
 
 RUN mkdir /src/build && \
     cd /src/build && \
-    cmake -G 'Ninja' .. && \
+    export PATH="$PATH:/usr/lib/ninja-build/bin/" && \
+    CXXFLAGS="-D__ptrace_request=int" cmake -G 'Ninja' .. && \
     cmake --build . && \
     cmake --build . --target install
 
-FROM debian:bookworm-slim
+FROM alpine:3.20
 
 COPY --from=builder /usr/local/bin/kcov* /usr/local/bin/
 COPY --from=builder /usr/local/share/doc/kcov /usr/local/share/doc/kcov
 
 RUN set -eux; \
-    apt-get update; \
-    apt-get install -y --no-install-recommends --no-install-suggests \
-        libcurl4 \
-        libdw1 \
-        zlib1g \
+    apk add --update --no-cache \
+        bash \
+        libcurl \
+        libdw \
+        zlib \
+        libgcc \
+        libstdc++ \
     ; \
-    apt-get clean; \
-    rm -rf /var/lib/apt/lists/*; \
     # Write a test script
     echo '#!/usr/bin/env bash\nif [[ true ]]; then\necho "Hello, kcov!"\nfi' > /tmp/test-executable.sh; \
```

Comment:
The ' probably messes this up. I think the best might be to use multi-line-syntax, e.g., something like

```
cat <<EOF > /tmp/test-executable.sh
#!/usr/bin/env bash
if [[ true ]]; then
    echo "Hello, kcov!"
fi
EOF
```

Reply:
ah yes !
Good catch, I need to add -e to echo.

the EOF stuff does not work fine in the Dockerfile syntax..